### PR TITLE
Add multiple PK support for SQLite and Postgres.

### PR DESCRIPTION
**Pull request by turnkey-commerce**
This release addresses issues #111 and #112.  It builds on the methodology started by stub42 with the following enhancements:
  - backward compatible for existing templates by preserving the old PrimaryKey in all cases for the templates.
  - adds template and related function support for Postgres, MySQL, and SQLite.
I encountered some difficulties in the functions that are used by the templates in the support of the variadic argument. To address that I added some functions suffixed by "multi" that use the column slice rather than the variadic for the input.  It was necessary to provide anotherr function anyway in the case of numbering some of the column inputs to Postgres.

### CODE REVIEW

File: templates/mysql.type.go.tpl [hunk 1]
```diff
@@ -98,15 +98,27 @@ func ({{ $short }} *{{ .Name }}) Insert(db XODB) error {
...
-		const sqlstr = `UPDATE {{ $table }} SET ` +
-			`{{ colnamesquery .Fields ", " .PrimaryKey.Name }}` +
-			` WHERE {{ colname .PrimaryKey.Col }} = ?`
-
-		// run query
-		XOLog(sqlstr, {{ fieldnames .Fields $short .PrimaryKey.Name }}, {{ $short }}.{{ .PrimaryKey.Name }})
-		_, err = db.Exec(sqlstr, {{ fieldnames .Fields $short .PrimaryKey.Name }}, {{ $short }}.{{ .PrimaryKey.Name }})
-		return err
+		{{ if gt ( len .PrimaryKeyFields ) 1 }}
+			// sql query with multiple PK's
```
Comment (stub42) [inline templates/mysql.type.go.tpl]: Its a multi part primary key. Multiple primary keys is an oxymoron.

Comment (turnkey-commerce) [inline templates/mysql.type.go.tpl]: Good point thanks, I was being a little loose with the terminology there, will fix it later.

File: templates/postgres.type.go.tpl [hunk 2]
```diff
@@ -87,17 +87,30 @@ func ({{ $short }} *{{ .Name }}) Insert(db XODB) error {
...
+			const sqlstr = `UPDATE {{ $table }} SET (` +
+				`{{ colnames .Fields .PrimaryKey.Name }}` +
+				`) = ( ` +
+				`{{ colvals .Fields .PrimaryKey.Name }}` +
+				`) WHERE {{ colname .PrimaryKey.Col }} = ${{ colcount .Fields .PrimaryKey.Name }}`
+
+			// run query
+			XOLog(sqlstr, {{ fieldnames .Fields $short .PrimaryKey.Name }}, {{ $short }}.{{ .PrimaryKey.Name }})
+			_, err = db.Exec(sqlstr, {{ fieldnames .Fields $short .PrimaryKey.Name }}, {{ $short }}.{{ .PrimaryKey.Name }})
+		{{ end }}
```
Comment (turnkey-commerce) [inline templates/postgres.type.go.tpl]: Missing "return err" line before the {{ end }} statement, should be:
```
			_, err = db.Exec(sqlstr, {{ fieldnames .Fields $short .PrimaryKey.Name }}, {{ $short }}.{{ .PrimaryKey.Name }})
			return err
 		{{ end }}`
```

**Comment by turnkey-commerce** [comment]
Replaced by PR #139

**Pull request closed without merging**